compiler/ast.nim

```diff
@@ -1834,6 +1834,7 @@ proc toVar*(typ: PType; kind: TTypeKind; idgen: IdGenerator): PType =
 proc toRef*(typ: PType; idgen: IdGenerator): PType =
   ## If ``typ`` is a tyObject then it is converted into a `ref <typ>` and
   ## returned. Otherwise ``typ`` is simply returned as-is.
+  result = typ
```

Comment:
seems correct from doc comment but was previous behavior (nil) relied on?
is there any test case demonstrating the bugfix?

Reply:
`toRef` is used in two places for `excType`. 

For example, it is used to convert `ObjException ` to `ref ObjException`. It doesn't matter whether it is orginal type or nil I think. Previous behavior (nil) seems not be relied on.
```nim
type
  ObjException = object of Exception

try:
  echo "2233"
except ObjException as e:
  echo "3322"
```